phockup.py

```diff
@@ -62,6 +63,11 @@ def main(argv):
             timestamp = True
             printer.line("Using file's timestamp")
         
+        if opt in ("-x", "--threads"):
+            if not arg:
+                printer.error("Number of threads cannot be empty")
+            threads = int(arg)
```

Comment:
Maybe we should limit the maximum threads to the CPU cores count?

Reply:
I thought about it, but the 1000 file test gave better results using 8 threads on my machine with just 4 core cpu. Maybe it's because the bottle neck is i/o, not the atual processing, I don't know. Please, let me know if you also see improvements with more threads than cores on your machine for large inputs.

Reply:
I guess you have 4 real cores but the CPU has 4 virtual cores making the total of 8. Check your output of `os.cpu_count()`

Reply:
This command returns 4

Reply:
Can you please tell me the model of your cpu

Reply:
With `cat  /proc/cpuinfo` I got:
> model name	: Intel(R) Core(TM) i5 CPU       M 460  @ 2.53GHz

This info is shown for 4 processors, being `core id 0` with `2 cpu cores` and `core id 2` with `2 cpu cores`.

Reply:
I think limiting to the maximum threads to CPU is not a good thing, since I/O is the bottle neck here. There's some discussion about it in: https://stackoverflow.com/questions/481970/how-many-threads-is-too-many . But the point is that threads block on I/O, and this happens each time `exiftool` is called. So it's good to have another thread ready to process right after a I/O block. Or you'll ended up with the cpu not being used 100% of the time.

Reply:
I hard coded a 16 threads limit, but now I guess even that would not be a good maximum for every machine. The link I mentioned suggest not limiting it at all,  and just test and see which value works better for each scenario.

Reply:
On my machine I got best results with 4 threads for small sets (< 50 files) and 8 threads for big sets (1000), while 16 threads don't perform any better in my tests.

But when I tested with exiftool batch input (not included in this code), the best results came using only 2 threads. I guess it's because in this case we don't have a I/O bottle neck anymore, and processing speed gain is limited since python threads implementation has _global interpreter lock_.

Reply:
Based on what I said here, I'll remove the hard coded threads maximum limit (16) in the next comit.